projects/hslayers-sensors/src/components/sensors/sensors-unit-dialog.component.ts

```diff
@@ -103,29 +88,32 @@ export class HsSensorsUnitDialogComponent
 
   /**
    * @param interval - Clicked interval button
+   * @param generate - Controling wether to fetch observations and generate charts as well. Not necessary on init
    * Get data for different time interval and regenerate
    * chart
    */
-  timeButtonClicked(interval): void {
+  timeButtonClicked(interval, generate = true): void {
     this.hsSensorsUnitDialogService.get(this.data.app).currentInterval =
       interval;
     const fromTo = this.hsSensorsUnitDialogService.getTimeForInterval(interval);
     Object.assign(this.customInterval, {
       fromTime: fromTo.from_time.toDate(),
       toTime: fromTo.to_time.toDate(),
     });
-    this.hsSensorsUnitDialogService
-      .getObservationHistory(
-        this.hsSensorsUnitDialogService.get(this.data.app).unit,
-        interval,
-        this.data.app
-      )
-      .then((_) => {
-        this.hsSensorsUnitDialogService.createChart(
-          this.hsSensorsUnitDialogService.get(this.data.app).unit,
+    if (generate) {
+      this.hsSensorsUnitDialogService
+        .getObservationHistory(
+          this.hsSensorsUnitDialogService.get(this.data.app).unit[0], //ALWAYS JUST ONE?
```

Comment:
```suggestion
          this.hsSensorsUnitDialogService.get(this.data.app).unit[0], //FIXME: ALWAYS JUST ONE?
```

Reply:
Thanks for pointing these out, forgot to get back and fix it. Its now done in [6c8dfc8](https://github.com/hslayers/hslayers-ng/pull/4100/commits/6c8dfc86d7c3ff075f58f331acd3a605cedee086) and [32811e8](https://github.com/hslayers/hslayers-ng/pull/4100/commits/32811e81dde8c5c5931a3705c58f79a8d5e1d4b0)

Reply:
Great, thanks!